beacon-chain/blockchain/chain_info.go

```diff
@@ -288,23 +283,7 @@ func (s *Service) IsCanonical(ctx context.Context, blockRoot [32]byte) (bool, er
 // ChainHeads returns all possible chain heads (leaves of fork choice tree).
 // Heads roots and heads slots are returned.
 func (s *Service) ChainHeads() ([][32]byte, []types.Slot) {
-	nodes := s.ProtoArrayStore().Nodes()
```

Comment:
Don't we still need to keep the old behavior when feature flag is not applied?

Reply:
This is an example of better encapsulation I refer above: the blockchain package does not need to know how the forkchoice package computes the head. It should request them. Each different store will implement the function Tips() the way it's convenient to them.